lib/transcoder_web/live/upload_live.ex

```diff
@@ -0,0 +1,58 @@
+defmodule TranscoderWeb.UploadLive do
+  alias Transcoder.Model.Video
+  use TranscoderWeb, :live_view
+
+  @impl Phoenix.LiveView
+  def mount(_params, _session, socket) do
+    {
+      :ok,
+      socket
+      |> assign(:transcoded_videos, [])
+      |> allow_upload(:video, accept: ~w(video/*), max_file_size: 1_000_000_000)
+    }
+  end
+
+  @impl Phoenix.LiveView
+  def handle_event("validate", _params, socket) do
+    {:noreply, socket}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_event("cancel-upload", %{"ref" => ref}, socket) do
+    {:noreply, cancel_upload(socket, :video, ref)}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_event("save", _params, socket) do
+    consume_uploaded_entries(socket, :video, &consume_upload/2)
+
+    {:noreply,
+     socket
+     |> put_flash(:info, "Transcoding scheduled")
+     |> assign(:transcoded_videos, [])}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_info({:transcode_result, video}, socket) do
+    {:noreply, update(socket, :transcoded_videos, &[video | &1])}
+  end
+
+  defp consume_upload(%{path: path}, entry) do
+    [_, extension] = String.split(entry.client_name, ".")
+    filename = Path.basename(path) <> "." <> extension
+    video = %Video{filename: filename}
+    File.cp!(path, Video.path(video))
+    {:ok, pid} = Transcoder.start_link()
+
+    for resolution <- [:"360p", :"480p", :"720p"] do
```

Comment:
Actually, it's not forbidden, but among elixir devs it's considered a bad practice to use atoms like that. Either use strings or atoms without `:"`

Reply:
Had to quote since they start with number.

Reply:
I see, I think it's okay to leave them as they are. Atoms are now typed, so we get compile-time validations for free in comparison to strings. If we were talking a few weeks ago, I'd recommend using strings here, since they're more readable, now - atoms are great! 💯